### PR TITLE
Handle raw messages instead of processed messages

### PR DESCRIPTION
**Pull request by steelbrain**
Linter will no longer be keeping UI packages in a fairy world, it now hands over them the raw messages it gets from providers

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -18,3 +20,13 @@ export function getEditorsMap(editorUI) {
   })
   return {editors, diff, activePaths}
 }
+
+export function getMessageRange(message) {
+  return message.range ? (
+    message.range.constructor.name === 'Array' ? Range.fromObject(message.range) : message.range
```
Comment (Arcanemagus) [inline lib/helpers.js]: `Range.fromObject` can accept a `Range`, in which case it just returns it. You might as well just always call that.

(Although this _does_ avoid a function call...)

File: lib/helpers.js [hunk 2]
```diff
@@ -18,3 +20,13 @@ export function getEditorsMap(editorUI) {
...
+  ) : null
+}
+
+export function getMessageClass(message) {
+  return message.class + ' ' + message.type.toLowerCase()
```
Comment (Arcanemagus) [inline lib/helpers.js]: Shouldn't this be `message.className` due to https://github.com/atom-community/linter/commit/87e0f4dc24d142935001f8f45256e84f04fefd3a?

**Comment by Arcanemagus** [comment]
I really think that UI services _should_ live in a fairy world... Linter should be validating messages and getting them into a consistent state before they are sent off to the UI service to be rendered.

File: lib/helpers.js (hunk 2 above)
Comment (steelbrain) [inline lib/helpers.js]: It was reverted :wink:

**Comment by steelbrain** [comment]
I would love to do that and I have been doing all that all this time, but it's quite pricey. We have to iterate over all of the messages and all of the traces of those messages, even when we don't need it. IMO these few lines once to avoid all the ops is worth it

**Pull request merged**